pkg/resources/kafka/kafka.go

```diff
@@ -802,7 +802,16 @@ func (r *Reconciler) reconcileKafkaPvc(log logr.Logger, brokersDesiredPvcs map[s
 					if err := patch.DefaultAnnotator.SetLastAppliedAnnotation(desiredPvc); err != nil {
 						return errors.WrapIf(err, "could not apply last state to annotation")
 					}
-					desiredPvc = currentPvc
+
+					if isDesiredStorageValueInvalid(desiredPvc, currentPvc) {
+						return errorfactory.New(errorfactory.InternalError{}, nil, "one can not reduce the size of a PVC", "kind", desiredType)
+					}
+
+					storageSize := desiredPvc.Spec.Resources.Requests.Storage().Value()
+					// because we are using generated names
```

Comment:
wouldn't `desiredopvc = currentPvc.DeepCopy()` copy all fields?

Reply:
I don't think we need DeepCopy, I think copying spec is enough.